src/udev/net/link-config.c

```diff
@@ -380,11 +381,17 @@ int link_config_apply(link_config_ctx *ctx, link_config *config,
         if (!old_name)
                 return -EINVAL;
 
-        r = ethtool_set_speed(&ctx->ethtool_fd, old_name, config->speed / 1024, config->duplex);
-        if (r < 0)
-                log_warning_errno(r, "Could not set speed or duplex of %s to %zu Mbps (%s): %m",
-                                  old_name, config->speed / 1024,
-                                  duplex_to_string(config->duplex));
+        r = ethtool_set_glinksettings(&ctx->ethtool_fd, old_name, config->speed / 1000000, config->duplex, config->autoneg);
+        if (r < 0) {
+
+                if (r == -EOPNOTSUPP)
+                        r = ethtool_set_speed(&ctx->ethtool_fd, old_name, config->speed / 1000000, config->duplex);
```

Comment:
i wonder if we shouldn't round up here? i.e. change this to `(config->speed + 999999) / 1000000)`?

(i am just asking this, i really don't know whether rounding up or down is more appropriate here...)